sim/paladin/judgement.go

```diff
@@ -164,6 +164,34 @@ func (paladin *Paladin) CanJudgementOfWisdom(sim *core.Simulation) bool {
 	return paladin.canJudgement(sim) && paladin.CurrentSeal == paladin.SealOfWisdomAura
 }
 
+// Defines judgement refresh behavior from attacks
+// Returns extra mana if a different pally applied Judgement of Wisdom
+func (paladin *Paladin) setupJudgementRefresh() {
+	const mana = 74 / 2
+	paladin.RegisterAura(core.Aura{
+		Label:    "Refresh Judgement",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHit: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, spellEffect *core.SpellEffect) {
+			if spellEffect.Landed() && spellEffect.ProcMask.Matches(core.ProcMaskMeleeWhiteHit) {
+				if paladin.CurrentJudgement != nil && paladin.CurrentJudgement.IsActive() {
+					// Refresh the judgement
+					paladin.CurrentJudgement.Refresh(sim)
+
+					// Check if current judgement is not JoW and also that JoW is on the target
+					if paladin.CurrentJudgement.ActionID.SpellID != 27164 &&
```

Comment:
This could be written more clearly as `paladin.CurrentJudgement != paladin.JudgementOfWisdomAura && paladin.JudgementOfWisdomAura.IsActive()`